test/common/common/utility_test.cc

```diff
@@ -116,6 +118,65 @@ TEST(DateUtil, NowToMilliseconds) {
   EXPECT_EQ(12345067, DateUtil::nowToMilliseconds(test_time));
 }
 
+TEST(OutputBufferStream, FailsOnWriteToEmptyBuffer) {
+  constexpr char data = 'x';
+  OutputBufferStream ostream{nullptr, 0};
+  ASSERT_TRUE(ostream.good());
+
+  ostream << data;
+
+  EXPECT_TRUE(ostream.bad());
+}
+
+TEST(OutputBufferStream, CanWriteToBuffer) {
+  constexpr char data[] = "123";
+  std::array<char, 3> buffer;
+
+  OutputBufferStream ostream{buffer.data(), buffer.size()};
+  ostream << data;
+
+  EXPECT_EQ(std::string(buffer.data(), ostream.numValidBytesWrittenToBuffer()), data);
+}
+
+TEST(OutputBufferStream, CannotOverwriteBuffer) {
+  constexpr char data[] = "123";
+  std::array<char, 2> buffer;
+
+  OutputBufferStream ostream{buffer.data(), buffer.size()};
+  ostream << data << std::endl;
+
+  EXPECT_EQ(std::string(buffer.data(), ostream.numValidBytesWrittenToBuffer()), "12");
+}
+
+TEST(OutputBufferStream, DoesNotAllocateMemoryEvenIfWeTryToOverflowBuffer) {
+  constexpr char data[] = "123";
+  std::array<char, 2> buffer;
+  Stats::TestUtil::MemoryTest memory_test;
+
+  OutputBufferStream ostream{buffer.data(), buffer.size()};
+  ostream << data << std::endl;
+
+  EXPECT_EQ(memory_test.consumedBytes(), 0);
+  EXPECT_EQ(std::string(buffer.data(), ostream.numValidBytesWrittenToBuffer()), "12");
+}
+
+TEST(OutputBufferStream, ReturnsNumberOfBytesWrittenIntoBuffer) {
+  constexpr char data[] = "123";
+  std::string buffer;
+  buffer.resize(3);
```

Comment:
I'm not sure how portable it is to write to std::string representation as done in this test.  It's possible that newer C++ standards banned std::string implementations that would make this kind of direct manipulation invalid.

Reply:
replaced with `std::array`